### PR TITLE
Changing preprocessor unrolls to pragma in ANR

### PR DESCRIPTION
**Pull request by aaronkintel**
# Existing Sample Changes
## Description

I changed preprocessor unrolls to `#pragma unroll` when the compiler uses `ivdep(array)`.

Fixes Issue# 

Compiler could not achieve II=1 for some loops since `ivdep(array)` could not penetrate lambdas from the unroller.

## External Dependencies

List any external dependencies created as a result of this change.

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Implement fixes for ONSAM Jiras

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Compiles to reporting and simulator with command line. Previously the compilation failed.

- [x] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode
- [ ] When compiling the compliler flag "-Wall -Wformat-security -Werror=format-security" was used

### CODE REVIEW

File: DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/anr/src/row_stencil.hpp [hunk 1]
```diff
@@ -96,20 +96,20 @@ void RowStencil(IndexT rows, IndexT cols, const InType zero_val,
   const IndexT col_loop_bound = padded_cols / parallel_cols;
 
   // the shift register
-  [[intel::fpga_register]]
-  fpga_tools::ShiftReg<InType, kShiftRegSize> shifty_pixels;
+  [[intel::fpga_register]] fpga_tools::ShiftReg<InType, kShiftRegSize>
```
Comment (mendell27) [inline DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/anr/src/row_stencil.hpp]: Please use  // NO-FORMAT: Attribute

**Review by mendell27 — APPROVED** [review]

File: DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/anr/src/column_stencil.hpp [hunk 2]
```diff
@@ -210,4 +216,4 @@ void ColumnStencil(IndexT rows, IndexT cols, const InType zero_val,
   }
 }
 
-#endif /* __COLUMN_STENCIL_HPP__ */
\ No newline at end of file
+#endif /* __COLUMN_STENCIL_HPP__ */
```
Comment (mendell27) [inline DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/anr/src/column_stencil.hpp:219]: Where did this come from?

Comment (mendell27) [reply DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/anr/src/column_stencil.hpp:219]: It sort of looks the same

**Review by mendell27 — APPROVED** [review]

File: DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/anr/src/column_stencil.hpp (hunk 2 above)
Comment (aaronkintel) [reply DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/anr/src/column_stencil.hpp:219]: I am not sure even what it means. I don't have a newline at the end.

**Pull request closed without merging**